tests/std/tests/P0896R4_ranges_alg_uninitialized_move/test.cpp

```diff
@@ -0,0 +1,218 @@
+            not_ranges_destroy(wrapped_output);
+            if (!is_constant_evaluated()) {
+                assert(int_wrapper::constructions == 3);
+                assert(int_wrapper::destructions == 3);
+            }
+        }
+
+        { // Validate iterator overload
+            int_wrapper input[3] = {13, 55, 12345};
+            int_wrapper::clear_counts();
+            R wrapped_input{input};
+            holder<int_wrapper, 3> mem;
+            W wrapped_output{mem.as_span()};
+
+            const same_as<uninitialized_move_result<iterator_t<R>, iterator_t<W>>> auto result = uninitialized_move(
+                wrapped_input.begin(), wrapped_input.end(), wrapped_output.begin(), wrapped_output.end());
+            if (!is_constant_evaluated()) {
+                assert(int_wrapper::constructions == 3);
+                assert(int_wrapper::destructions == 0);
+            }
+            assert(result.in == wrapped_input.end());
+            assert(result.out == wrapped_output.end());
+            assert(equal(wrapped_output, expected_output, equal_to{}, &int_wrapper::val));
+            assert(equal(input, expected_input, equal_to{}, &int_wrapper::val));
+            not_ranges_destroy(wrapped_output);
+            if (!is_constant_evaluated()) {
+                assert(int_wrapper::constructions == 3);
+                assert(int_wrapper::destructions == 3);
+            }
+        }
+    }
+};
+
+struct throwing_test {
+    static constexpr int expected_input[] = {-1, -1, int_wrapper::magic_throwing_val, 12345};
+
+    template <ranges::input_range R, ranges::forward_range W>
+    static void call() {
+        // Validate only range overload (one is plenty since they are backends)
+        int_wrapper input[] = {13, 55, int_wrapper::magic_throwing_val, 12345};
+        int_wrapper::clear_counts();
+        R wrapped_input{input};
+        holder<int_wrapper, 4> mem;
+        W wrapped_output{mem.as_span()};
+
+        try {
+            (void) ranges::uninitialized_move(wrapped_input, wrapped_output);
+            assert(false);
+        } catch (int i) {
+            assert(i == int_wrapper::magic_throwing_val);
+        } catch (...) {
+            assert(false);
+        }
+        assert(int_wrapper::constructions == 2);
+        assert(int_wrapper::destructions == 2);
+        assert(ranges::equal(input, expected_input, ranges::equal_to{}, &int_wrapper::val));
+    }
+};
+
+template <test::ProxyRef IsProxy>
+using test_input  = test::range<test::input, int_wrapper, test::Sized::no, test::CanDifference::no, test::Common::no,
+    test::CanCompare::yes, IsProxy>;
+using test_output = test::range<test::fwd, int_wrapper, test::Sized::no, test::CanDifference::no, test::Common::no,
+    test::CanCompare::yes, test::ProxyRef::no>;
+
+constexpr void run_tests() {
+    // The algorithm is oblivious to non-required category, size, difference, and requires non-proxy references for the
+    // output range.

```

Comment:
This is hard to parse (I think because there's a list of "category, size, difference" without an "and", followed by a separate "and" clause).